dkc/core/rest/authorized_upload.py

```diff
@@ -0,0 +1,113 @@
+from datetime import timedelta
+import logging
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.core.signing import BadSignature, Signer
+from django.shortcuts import get_object_or_404
+from django.template.loader import render_to_string
+from django.utils import timezone
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import mixins, serializers
+from rest_framework.decorators import action
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import View
+from rest_framework.viewsets import GenericViewSet
+
+from dkc.core.models import AuthorizedUpload, Folder
+from dkc.core.permissions import Permission
+
+
+class AuthorizedUploadSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = AuthorizedUpload
+        fields = ['id', 'created', 'creator', 'expires', 'folder', 'signature']
+
+    signature: str = serializers.SerializerMethodField()
+
+    def get_signature(self, upload: AuthorizedUpload) -> str:
+        return Signer().sign(str(upload.id))
```

Comment:
It occurs to me that there's a slight problem with just signing the ID without scoping it to a specific model type -- if the server ever started signing IDs for other models, someone could pretend that a signature for model foo with ID 123 was a signature for model bar with ID 123 and the server would accept it. So I think I will change this to either use `signing.dumps` and include the model, or else sign a string like e.g. `'AuthorizedUpload.123'`.

Reply:
The general principle here is that any features making use of the same signing key should have a disjoint set of possible values to be signed.